src/base/bittorrent/session.cpp

```diff
@@ -2146,12 +2146,21 @@ bool Session::addTorrent_impl(const std::variant<MagnetUri, TorrentInfo> &source
         }
 
         Q_ASSERT(p.file_priorities.empty());
-        std::transform(addTorrentParams.filePriorities.cbegin(), addTorrentParams.filePriorities.cend()
-                       , std::back_inserter(p.file_priorities), [](const DownloadPriority priority)
+        if (addTorrentParams.filePriorities.size() == 0)
         {
-            return static_cast<lt::download_priority_t>(
+            // Use qBittorrent default priority rather than libtorrent's (4)
+            p.file_priorities.resize(metadata.filesCount(), static_cast<lt::download_priority_t>(
```

Comment:
Can you replace it with an assignment instead? I feel it is wrong to have resize semantics here.
For example:
```c++
p.file_priorities = QVector(metadata.filesCount(), ...);
```
Same to the other code below.

Reply:
Done, but it's a `std::vector` not a `QVector`